cogs/twitter_link_fixer.py

```diff
@@ -0,0 +1,28 @@
+import re
+from urllib.parse import urlparse
+
+from discord.ext import commands
+
+
+class TwitterLinkFixerCog(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    def rewrite_url(self, message):
+        parse_url = urlparse(message)
+        return parse_url._replace(netloc="fxtwitter.com").geturl()
+
+    def find_twitter_urls(self, message):
+        twitter_or_x_pattern = re.compile(r'https?://(?:www\.)?(?:twitter\.com|x\.com)/\S+')

```

Comment:
nitpick: could make the regex case-insensitive like `https?://(?:www\.)?(?:twitter\.com|x\.com)/\S+/i`

Reply:
nvm, just realized you're lowering the case for all calls.